packages/scripts/src/create-webpack-configs.ts

```diff
@@ -179,6 +184,20 @@ export function createWebpackConfig({
                 ]
             );
         }
+        if (webpackHot) {
+            // we need the hot middleware client into the env entry
+            entry[envName] = ['webpack-hot-middleware/client', entryPath];
+            plugins.push(new webpack.HotModuleReplacementPlugin());
+            plugins.push(
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+                new ReactRefreshWebpackPlugin({
```

Comment:
@AviVahl @RomanYarik  I'm not really sure about this plugin here
everything I did up to this point is related to webpack hot reloading in general, and is only applicable to applications which have the flag turned on and a web env, so I feel fine about that - along with the fact that we actually are opinionated about webpack

This plugin is only applicable for react applications and should run in conjuncture with the babel/typescript transformation in the users development webpack config, and it feels a bit too opinionated on out part to set it up for them

I would consider documenting something in the lines of
To set up webpack hot reloading for your project you should
1. add ReactRefreshWebpackPlugin to your webpack config
2. Add the babel/typescript transformation to your webpack config
3. engineer start --webpackHot true

What do you think?

Reply:
Actually while adding this comment I decided to move on with it, in the next commit I moved the react part of hot reloading to the responsibility of the user - that way the can decide the hot reloading solution, we just add the webpack infra for them